modules/setting/project.go

```diff
@@ -12,8 +44,58 @@ var (
 		ProjectBoardBasicKanbanType: []string{"To Do", "In Progress", "Done"},
 		ProjectBoardBugTriageType:   []string{"Needs Triage", "High Priority", "Low Priority", "Closed"},
 	}
+	ProjectAutomation = struct {
+		Enabled            bool
+		MaxRulesPerProject int
+	}{
+		Enabled:            false,
+		MaxRulesPerProject: 25,
+	}
+	ProjectAutomationKanban = ProjectAutomationConfig{
+		Rule1:  []string{"trigger=status:closed", "action=move:Done"},
+		Rule2:  []string{"context=Done", "trigger=status:reopened", "action=move:To Do"},
+		Rule3:  []string{"trigger=approve", "target=pr", "action=move:Done"},
+		Rule4:  []string{"trigger=xref:closes", "target=pr", "action=assign_project"},
+		Rule5:  []string{"trigger=assign_project", "target=pr", "action=move:To Do"},
+		Rule6:  []string{"trigger=move:To Do", "action=unassign"},
+		Rule7:  []string{"trigger=move:To Do", "action=status:reopened"},
+		Rule8:  []string{"trigger=move:To Do", "action=unassign_reviewers"},
+		Rule9:  []string{"trigger=move:In Progress", "action=assign"},
+		Rule10: []string{"trigger=move:In Progress", "action=status:reopened"},
+		Rule11: []string{"trigger=move:In Progress", "action=assign_reviewer"},
+		Rule12: []string{"trigger=move:Done", "action=status:closed", "target=issue"},
+		Rule13: []string{"trigger=move:Done", "action=approve", "target=pr"},
+		Rule14: []string{"context=To Do", "trigger=assign", "target=issue", "action=move:In Progress"},
+		Rule15: []string{"context=To Do", "trigger=assign_reviewer", "action=move:In Progress"},
+		Rule16: []string{"context=In Progress", "trigger=xref:closes", "target=issue", "action=label:Status/Blocked"},
+		Rule17: []string{"context=In Progress", "trigger=approve", "target=issue", "action=unlabel:Status/Blocked"},
+	}
```

Comment:
??

Reply:
Care to elaborate?

I assume I would need to codify the default automation rules _somewhere_ and _somehow_.

Given that this is the file where the default configuration for column titles are, I at least think I figured out the _somewhere_.

Suggestions on how to simplify the _somehow_ are very much welcome.

As for the default rules themselves, they are up for debate.  I threw a few things at the proverbial wall that I thought might be useful, but the overall theme for the intended workflow is:
1. assign person to issue ("In progress")
2. person creates PR closing the issue (this "blocks" the issue and adds the PR to the project as a separate ticket)
3. someone else reviews and approves the PR (which "unblocks" the issue) (the PR is "Done", the related issue is still "In progress", but not blocked)
4. the issue gets merged ("Done")

The default rules also support being triggered by moving the issues directly on the project board thus changing the issue status and assigning people (not sure what to call that, but lets say it is a "proactive workflow") as well as moving the issues whenever people are assigned or the status changes etc (lets call that a "reactive workflow").  Thus many rules are "duplicated" in order to support both types of workflows.